apps/cowswap-frontend/src/modules/appData/utils/buildAppDataHooks.ts

```diff
@@ -0,0 +1,8 @@
+import { AppDataHooks, PostHooks, PreHooks } from '../types'
+
+export function buildAppDataHooks(pre: PreHooks, post: PostHooks): AppDataHooks {
```

Comment:
Why not using Omit<OrderInteractionHooks, 'version'> or even better, use the latest version of appData, where we shouldn't have the `version` so you can use `OrderInteractionHooks` directly.

Also, why passing an empty `post` array? i think its optional, right?

Reply:
Makes sense to both be optional.
As for the input being the same type as the output, I don't see much point in having the fn then. It would be this:

const hooks = buildAppDataHooks({pre:[data]})

// hooks == { pre: [ ...data ] }